File: 7-kyu/xmas-tree/python/solution.py
def xmastree(n):
    ret = [('_' * (n - i)) + ('#' * (i * 2 - 1)) + ('_' * (n - i)) for i in range(1, n + 1)]
    ret.append(('_' * (n - 1)) + '#' + ('_' * (n - 1)))
    ret.append(('_' * (n - 1)) + '#' + ('_' * (n - 1)))
    return ret

File: 7-kyu/xmas-tree/javascript/solution.js
const xMasTree = (n) => {
  let ret = [];
  for (let i = 1; i <= n; i++) {
    const underscores = "_".repeat(n - i);
    const center = "#".repeat(i * 2 - 1);
    ret.push(underscores + center + underscores);
  }
  const underTree = "_".repeat(n - 1) + "#" + "_".repeat(n - 1);
  for (let i = 0; i < 2; i++) {
    ret.push(underTree);
  }
  return ret;
};
